Mathlib/Algebra/MvPolynomial/Equiv.lean

```diff
@@ -318,40 +318,257 @@ lemma optionEquivRight_X_none : optionEquivRight R S₁ (X none) = C Polynomial.
+    (hi : (finSuccEquivNth R p f).coeff i ≠ 0) :
+      totalDegree ((finSuccEquivNth R p f).coeff i) + i ≤ totalDegree f := by
+  have hf'_sup : ((finSuccEquivNth R p f).coeff i).support.Nonempty := by
+    rw [Finset.nonempty_iff_ne_empty, ne_eq, support_eq_empty]
+    exact hi
+  -- Let σ be a monomial index of ((finSuccEquivNth R p f).coeff i) of maximal total degree
+  have ⟨σ, hσ1, hσ2⟩ := Finset.exists_mem_eq_sup (support _) hf'_sup
+                          (fun s => Finsupp.sum s fun _ e => e)
+  -- Then σ.insertNth p i is a monomial index of f with total degree equal to the desired bound
+  convert le_totalDegree (s := σ.insertNth p i) _
+  · rw [totalDegree, hσ2, sum_insertNth _ _ p, add_comm]
+  · rw [← support_coeff_finSuccEquivNth]
+    exact hσ1
+
+/-- The support of `finSuccEquivNth R p f` equals the support of `f` projected onto the `p`-th
+variable. -/
+theorem support_finSuccEquivNth (f : MvPolynomial (Fin (n + 1)) R) :
+    (finSuccEquivNth R p f).support = Finset.image (fun m : Fin (n + 1) →₀ ℕ => m p) f.support := by
+  ext i
+  rw [Polynomial.mem_support_iff, Finset.mem_image, Finsupp.ne_iff]
+  constructor
+  · rintro ⟨m, hm⟩
+    refine ⟨m.insertNth p i, ?_, insertNth_apply_same _ _ _⟩
+    rw [← support_coeff_finSuccEquivNth]
+    simpa using hm
+  · rintro ⟨m, h, rfl⟩
+    refine ⟨m.removeNth p, ?_⟩
+    rwa [← coeff, zero_apply, ← mem_support_iff, support_coeff_finSuccEquivNth,
+      insertNth_self_removeNth]
+
+theorem mem_support_finSuccEquivNth {f : MvPolynomial (Fin (n + 1)) R} {x} :
+    x ∈ (finSuccEquivNth R p f).support ↔ x ∈ (fun m : Fin (n + 1) →₀ _ ↦ m p) '' f.support := by
+  simpa using congr(x ∈ $(support_finSuccEquivNth f))
+
+theorem image_support_finSuccEquivNth {f : MvPolynomial (Fin (n + 1)) R} {i : ℕ} :
+    Finset.image (Finsupp.insertNth p i) (Polynomial.coeff ((finSuccEquivNth R p) f) i).support =
+      f.support.filter fun m => m p = i := by
+  ext m
+  rw [Finset.mem_filter, Finset.mem_image, mem_support_iff]
+  conv_lhs =>
+    congr
+    ext
+    rw [mem_support_iff, finSuccEquivNth_coeff_coeff, Ne]
+  constructor
+  · rintro ⟨m', ⟨h, hm'⟩⟩
+    simp only [← hm']
+    exact ⟨h, by rw [insertNth_apply_same]⟩
+  · intro h
+    use m.removeNth p
+    rw [← h.2, insertNth_removeNth]
+    simp [h.1]
+
+lemma mem_image_support_coeff_finSuccEquivNth {f : MvPolynomial (Fin (n + 1)) R} {i : ℕ} {x} :
+    x ∈ Finsupp.insertNth p i '' ((finSuccEquivNth R p f).coeff i).support ↔
+      x ∈ f.support ∧ x p = i := by
+  simpa using congr(x ∈ $image_support_finSuccEquivNth)
+
+lemma mem_support_coeff_finSuccEquivNth {f : MvPolynomial (Fin (n + 1)) R} {i : ℕ} {x} :
+    x ∈ ((finSuccEquivNth R p f).coeff i).support ↔ x.insertNth p i ∈ f.support := by
+  rw [← (Finsupp.insertNth_right_injective p).mem_finset_image (a := x),
+    image_support_finSuccEquivNth]
+  simp only [Finset.mem_filter, mem_support_iff, ne_eq, insertNth_apply_same, and_true]
```

Comment:
Can these two be merged into one `simp only`? (untested)

Reply:
Unfortunately no.